enterprise/cmd/frontend/internal/auth/openidconnect/middleware.go

```diff
@@ -85,16 +86,22 @@ func handleOpenIDConnectAuth(db database.DB, w http.ResponseWriter, r *http.Requ
 
 	// If the actor is authenticated and not performing an OpenID Connect flow, then proceed to
 	// next.
+	//
+	// If a sign-out cookie has been set during a previous sign-out request, remove it by setting MaxAge < 0.
 	if actor.FromContext(r.Context()).IsAuthenticated() {
+		if common.HasSignOutCookie(r) {
+			http.SetCookie(w, &http.Cookie{Name: common.SignoutCookie, Value: "", MaxAge: -1})
+		}
+
 		next.ServeHTTP(w, r)
 		return
 	}
 
 	// If there is only one auth provider configured, the single auth provider is OpenID Connect,
-	// and it's an app request, redirect to signin immediately. The user wouldn't be able to do
-	// anything else anyway; there's no point in showing them a signin screen with just a single
-	// signin option.
-	if ps := providers.Providers(); len(ps) == 1 && ps[0].Config().Openidconnect != nil && !isAPIRequest {
+	// it's an app request, and the sign-out cookie is not present, redirect to sign-in immediately.
+	//
+	// For sign-out requests (sign-out cookie is  present), the user is redirected to the Sourcegraph login page.
+	if ps := providers.Providers(); len(ps) == 1 && ps[0].Config().Openidconnect != nil && !common.HasSignOutCookie(r) && !isAPIRequest {
```

Comment:
I think this line has outgrown its one-lined-ness

```suggestion
	ps := providers.Providers()
	openIDConnectEnabled := len(ps) == 1 && ps[0].Config().Openidconnect != nil
	if  openIDConnectEnabled && !common.HasSignOutCookie(r) && !isAPIRequest {
```